src/components/base/base-textarea/__tests__/BaseTextarea.spec.ts

```diff
@@ -0,0 +1,135 @@
+import type { ComponentPublicInstance } from 'vue'
+import { mount } from '@vue/test-utils'
+import { describe, it, expect } from 'vitest'
+
+import BaseTextarea from '@/components/base/base-textarea/BaseTextarea.vue'
+import type { ITextareaComponent } from '@/components/base/base-textarea/types'
+
+describe('BaseTextarea', () => {
+  it('should update character count when input changes', async () => {
+    const wrapper = mount(BaseTextarea, {
+      props: {
+        id: 'textarea-1',
+        disabled: false
+      },
+      slots: {
+        label: 'Slot label',
+
+        error: `
+          
+          <span>error</span>
+        `,
+        message: 'this is my message'
+      }
+    })
+
+    const textarea = wrapper.find('textarea')
+    await textarea.setValue('Hello')
+    const vm = wrapper.vm as ComponentPublicInstance<any & { value: number }>
+
+    expect(vm.value.length).toBe(5)
+  })
+
+  describe('When there is an error', () => {
+    it('should show error message', async () => {
+      const wrapper = mount(BaseTextarea, {
+        slots: {
+          error: 'This is an error message'
+        },
+        props: {
+          required: true
+        }
+      })
+      const errorMessage = wrapper.find('[data-testID="ui-textarea-error"]')
+      expect(errorMessage.exists()).toBeTruthy()
+      expect(errorMessage.text()).toBe('This is an error message')
+    })
+  })
+
+  describe('when help button is clicked', () => {
+    it('should show help message ', async () => {
```

Comment:
estos describes `when help button is clicked`, `when clear button is clicked` se podrían unificar en `Test user events behaviours`